aioxmpp/avatar/service.py

```diff
@@ -509,19 +566,20 @@ def publish_avatar_set(self, avatar_set):
 
         id_ = avatar_set.png_id
 
-        yield from self._pubsub.publish(
-            None,
-            namespaces.xep0084_data,
-            avatar_xso.Data(avatar_set.image_bytes),
-            id_=id_
-        )
+        with (yield from self._lock):
```

Comment:
[XEP-0163 section 3]( https://xmpp.org/extensions/xep-0163.html#publish) is a bit unclear on whether we always get a notification:
> If the publication logic dictates that event notifications shall be sent, the account owner's server generates notifications and sends them to all appropriate entities as described in the Receiving Event Notifications section of this document, as well as to any of the account owner's available resources.

If we would get a notification about the data, it would be easy to check consistency. If the server is clever enough to not send the payload along (but only notify about the ID), this would be the desired behaviour.